### ISSUE TITLE
light curve data not found for some TOIs

### ISSUE THREAD

**Issue by avivajpeyi** [dfm/tess-atlas#157]
### Steps to reproduce: 
```python
>>> from tess_atlas.data import LightCurveData
>>> LightCurveData.from_database(tic=332579116, outdir='.')
...
ValueError: No light curves for TIC 332579116
```

### Expected outcome: 
```python
>>> LightCurveData.from_database(tic=72668830, outdir='.')
Warning: 30% (5871/19412) of the cadences will be ignored due to the quality mask (quality_bitmask=175).
<tess_atlas.data.lightcurve_data.LightCurveData object at 0x133b04eb0>
```

### TOIs with error

```
gen_24573525_0.log:ValueError: No light curves for TIC 332579116
gen_24573525_102.log:ValueError: No light curves for TIC 67535663
gen_24573525_103.log:ValueError: No light curves for TIC 266843799
gen_24573525_104.log:ValueError: No light curves for TIC 307079330
gen_24573525_10.log:ValueError: No light curves for TIC 37242226
gen_24573525_11.log:ValueError: No light curves for TIC 280304556
gen_24573525_12.log:ValueError: No light curves for TIC 200090347
gen_24573525_132.log:ValueError: No light curves for TIC 256722647
gen_24573525_133.log:ValueError: No light curves for TIC 47617038
gen_24573525_134.log:ValueError: No light curves for TIC 148833828
gen_24573525_135.log:ValueError: No light curves for TIC 178959058
gen_24573525_137.log:ValueError: No light curves for TIC 128531633
gen_24573525_13.log:ValueError: No light curves for TIC 409315469
gen_24573525_143.log:ValueError: No light curves for TIC 454038613
gen_24573525_14.log:ValueError: No light curves for TIC 385654601
gen_24573525_15.log:ValueError: No light curves for TIC 126864892
gen_24573525_168.log:ValueError: No light curves for TIC 413343957
gen_24573525_169.log:ValueError: No light curves for TIC 297361900
gen_24573525_16.log:ValueError: No light curves for TIC 32408928
gen_24573525_173.log:ValueError: No light curves for TIC 335040423
gen_24573525_174.log:ValueError: No light curves for TIC 450649506
gen_24573525_175.log:ValueError: No light curves for TIC 401980624
gen_24573525_176.log:ValueError: No light curves for TIC 315350812
gen_24573525_177.log:ValueError: No light curves for TIC 308191940
gen_24573525_178.log:ValueError: No light curves for TIC 343708863
gen_24573525_179.log:ValueError: No light curves for TIC 250306572
gen_24573525_17.log:ValueError: No light curves for TIC 409594381
gen_24573525_18.log:ValueError: No light curves for TIC 7274938
gen_24573525_193.log:ValueError: No light curves for TIC 369345064
gen_24573525_194.log:ValueError: No light curves for TIC 66620917
gen_24573525_195.log:ValueError: No light curves for TIC 429791387
gen_24573525_196.log:ValueError: No light curves for TIC 107440797
gen_24573525_197.log:ValueError: No light curves for TIC 341805908
gen_24573525_198.log:ValueError: No light curves for TIC 218483943
gen_24573525_199.log:ValueError: No light curves for TIC 379723086
gen_24573525_19.log:ValueError: No light curves for TIC 187861378
gen_24573525_1.log:ValueError: No light curves for TIC 459913687
gen_24573525_201.log:ValueError: No light curves for TIC 214243287
gen_24573525_20.log:ValueError: No light curves for TIC 108939482
gen_24573525_214.log:ValueError: No light curves for TIC 35315453
gen_24573525_218.log:ValueError: No light curves for TIC 79605791
gen_24573525_219.log:ValueError: No light curves for TIC 208297760
gen_24573525_21.log:ValueError: No light curves for TIC 142628514
gen_24573525_223.log:ValueError: No light curves for TIC 141412823
gen_24573525_224.log:ValueError: No light curves for TIC 263748033
gen_24573525_225.log:ValueError: No light curves for TIC 344600402
gen_24573525_226.log:ValueError: No light curves for TIC 326386495
gen_24573525_227.log:ValueError: No light curves for TIC 106940971
gen_24573525_22.log:ValueError: No light curves for TIC 114103136
gen_24573525_23.log:ValueError: No light curves for TIC 346995484
gen_24573525_24.log:ValueError: No light curves for TIC 334252687
gen_24573525_25.log:ValueError: No light curves for TIC 318054782
gen_24573525_262.log:ValueError: No light curves for TIC 461635621
gen_24573525_264.log:ValueError: No light curves for TIC 76420654
gen_24573525_265.log:ValueError: No light curves for TIC 8599009
gen_24573525_266.log:ValueError: No light curves for TIC 441589581
gen_24573525_26.log:ValueError: No light curves for TIC 157088184
gen_24573525_27.log:ValueError: No light curves for TIC 148667071
gen_24573525_28.log:ValueError: No light curves for TIC 141405814
gen_24573525_293.log:ValueError: No light curves for TIC 193336820
gen_24573525_294.log:ValueError: No light curves for TIC 298297931
gen_24573525_295.log:ValueError: No light curves for TIC 263218688
gen_24573525_296.log:ValueError: No light curves for TIC 40466976
gen_24573525_298.log:ValueError: No light curves for TIC 275758701
gen_24573525_299.log:ValueError: No light curves for TIC 436166416
gen_24573525_29.log:ValueError: No light curves for TIC 127328218
gen_24573525_2.log:ValueError: No light curves for TIC 220144363
gen_24573525_301.log:ValueError: No light curves for TIC 120183168
gen_24573525_302.log:ValueError: No light curves for TIC 406190563
gen_24573525_303.log:ValueError: No light curves for TIC 22069559
gen_24573525_30.log:ValueError: No light curves for TIC 99081059
gen_24573525_31.log:ValueError: No light curves for TIC 415143674
gen_24573525_32.log:ValueError: No light curves for TIC 319282294
gen_24573525_33.log:ValueError: No light curves for TIC 128127712
gen_24573525_34.log:ValueError: No light curves for TIC 127895118
gen_24573525_35.log:ValueError: No light curves for TIC 126357909
gen_24573525_36.log:ValueError: No light curves for TIC 113295476
gen_24573525_37.log:ValueError: No light curves for TIC 81055189
gen_24573525_38.log:ValueError: No light curves for TIC 157528286
gen_24573525_39.log:ValueError: No light curves for TIC 36227841
gen_24573525_3.log:ValueError: No light curves for TIC 53593542
gen_24573525_40.log:ValueError: No light curves for TIC 30825828
gen_24573525_41.log:ValueError: No light curves for TIC 24615998
gen_24573525_42.log:ValueError: No light curves for TIC 22121296
gen_24573525_43.log:ValueError: No light curves for TIC 19547831
gen_24573525_44.log:ValueError: No light curves for TIC 4102368
gen_24573525_45.log:ValueError: No light curves for TIC 445582110
gen_24573525_46.log:ValueError: No light curves for TIC 400683848
gen_24573525_47.log:ValueError: No light curves for TIC 393340920
gen_24573525_48.log:ValueError: No light curves for TIC 52928939
gen_24573525_49.log:ValueError: No light curves for TIC 291465998
gen_24573525_4.log:ValueError: No light curves for TIC 95057860
gen_24573525_50.log:ValueError: No light curves for TIC 286153328
gen_24573525_51.log:ValueError: No light curves for TIC 190986054
gen_24573525_52.log:ValueError: No light curves for TIC 269568703
gen_24573525_53.log:ValueError: No light curves for TIC 269249506
gen_24573525_54.log:ValueError: No light curves for TIC 192520375
gen_24573525_55.log:ValueError: No light curves for TIC 145038729
gen_24573525_56.log:ValueError: No light curves for TIC 145249556
gen_24573525_57.log:ValueError: No light curves for TIC 92744152
gen_24573525_58.log:ValueError: No light curves for TIC 82302240
gen_24573525_59.log:ValueError: No light curves for TIC 77102895
gen_24573525_5.log:ValueError: No light curves for TIC 237603062
gen_24573525_60.log:ValueError: No light curves for TIC 75208638
gen_24573525_61.log:ValueError: No light curves for TIC 74484886
gen_24573525_62.log:ValueError: No light curves for TIC 268643535
gen_24573525_63.log:ValueError: No light curves for TIC 74967390
gen_24573525_64.log:ValueError: No light curves for TIC 74197917
gen_24573525_65.log:ValueError: No light curves for TIC 53365065
gen_24573525_66.log:ValueError: No light curves for TIC 48276576
gen_24573525_67.log:ValueError: No light curves for TIC 45996325
gen_24573525_68.log:ValueError: No light curves for TIC 118308828
gen_24573525_69.log:ValueError: No light curves for TIC 442050698
gen_24573525_6.log:ValueError: No light curves for TIC 72494421
gen_24573525_70.log:ValueError: No light curves for TIC 80783538
gen_24573525_71.log:ValueError: No light curves for TIC 341272282
gen_24573525_72.log:ValueError: No light curves for TIC 178657161
gen_24573525_73.log:ValueError: No light curves for TIC 151920165
gen_24573525_74.log:ValueError: No light curves for TIC 146842334
gen_24573525_75.log:ValueError: No light curves for TIC 146664561
gen_24573525_76.log:ValueError: No light curves for TIC 133331642
gen_24573525_77.log:ValueError: No light curves for TIC 290388627
gen_24573525_78.log:ValueError: No light curves for TIC 363635669
gen_24573525_79.log:ValueError: No light curves for TIC 469811655
gen_24573525_7.log:ValueError: No light curves for TIC 158697949
gen_24573525_80.log:ValueError: No light curves for TIC 466682238
gen_24573525_81.log:ValueError: No light curves for TIC 466222485
gen_24573525_83.log:ValueError: No light curves for TIC 387193925
gen_24573525_84.log:ValueError: No light curves for TIC 384386061
gen_24573525_85.log:ValueError: No light curves for TIC 374348168
gen_24573525_86.log:ValueError: No light curves for TIC 343952976
gen_24573525_87.log:ValueError: No light curves for TIC 280726137
gen_24573525_88.log:ValueError: No light curves for TIC 310105752
gen_24573525_89.log:ValueError: No light curves for TIC 294097592
gen_24573525_8.log:ValueError: No light curves for TIC 280645209
gen_24573525_90.log:ValueError: No light curves for TIC 290407621
gen_24573525_91.log:ValueError: No light curves for TIC 278821913
gen_24573525_92.log:ValueError: No light curves for TIC 278540265
gen_24573525_93.log:ValueError: No light curves for TIC 278196907
gen_24573525_94.log:ValueError: No light curves for TIC 257520272
gen_24573525_95.log:ValueError: No light curves for TIC 255918903
gen_24573525_96.log:ValueError: No light curves for TIC 101696403
gen_24573525_97.log:ValueError: No light curves for TIC 30499203
gen_24573525_98.log:ValueError: No light curves for TIC 325527653
gen_24573525_9.log:ValueError: No light curves for TIC 281242007
```

**Comment by avivajpeyi** [comment]
`2118` TOIs have the above error

**Comment by avivajpeyi** [comment]
- We want to restrict to 2min cadence targets  
- these TOIs are probably found in full-frame images 

Check out the exo docs, only get data from SPOC

Dont want 20second cadence (too long!)

**Comment by avivajpeyi** [comment]
We are restricting only the 2min cadence targets already: 
https://github.com/dfm/tess-atlas/blob/f91b795da72fa9cab3f146636dcaff89c2e22d08/src/tess_atlas/data/lightcurve_data.py#L48

With #168 we are also forcing the use of only SPOC candidates + data.

**Comment by avivajpeyi** [comment]
From Dan: 
> yeah - you're definitely getting throttled. We probably need to put in some more friendly timeouts and back off strategies so that we don't piss off the archives. You could try catching that exception, adding a sleep for a minute and then trying again.

yeah Ill add in a sleep-and-try again. 

We've done it for getting other MAST data as well: 
https://github.com/dfm/tess-atlas/blob/e0c927969e35b7152404e958b7f565319c4d9c5a/src/tess_atlas/data/stellar_data.py#L128

I thought that `lk` had this built-in... Hmmm...

**Comment by avivajpeyi** [comment]
Taking a quick look a random TOI that failed: `TOI 164`

slurm log: 
```python
...
  File "/.../lightkurve/search.py", line 782, in search_lightcurve
    return _search_products(
  File "/.../lightkurve/search.py", line 930, in _search_products
    observations = _query_mast(
  File "/.../lightkurve/search.py", line 1128, in _query_mast
    obs = Observations.query_criteria(
  File "/.../astroquery/utils/class_or_instance.py", line 25, in f
    return self.fn(obj, *args, **kwds)
  File "/.../astroquery/utils/process_asyncs.py", line 26, in newmethod
    response = getattr(self, async_method_name)(*args, **kwargs)
  File "/.../astroquery/utils/class_or_instance.py", line 25, in f
    return self.fn(obj, *args, **kwds)
  File "/.../astroquery/mast/observations.py", line 283, in query_criteria_async
    position, mashup_filters = self._parse_caom_criteria(**criteria)
  File "/.../astroquery/mast/observations.py", line 160, in _parse_caom_criteria
    mashup_filters = self._portal_api_connection.build_filter_set("Mast.Caom.Cone",
  File "/.../astroquery/mast/discovery_portal.py", line 396, in build_filter_set
    self._get_col_config(service_name, fetch_name=column_config_name)
  File "/.../astroquery/mast/discovery_portal.py", line 234, in _get_col_config
    response = self._request("POST", self.COLUMNS_CONFIG_URL,
  File "/.../astroquery/mast/discovery_portal.py", line 182, in _request
    response = super(PortalAPI, self)._request(method, url, params=params, data=data,
  File "/.../astroquery/query.py", line 296, in _request
    response = query.request(self._session, stream=stream,
  File "/.../astroquery/query.py", line 67, in request
    return session.request(self.method, self.url, params=self.params,
  File "/.../requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/.../requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/.../requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='mast.stsci.edu', port=443): Max retries exceeded with url: /portal/Mashup/Mashup.asmx/columnsconfig (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x2b7e25f82c10>: Failed to establish a new connection: [Errno -2] Name or service not known'))
srun: error: john49: task 0: Exited with exit code 1
srun: Terminating job step 26224761.0
```

Hmm ok so `lk` is indeed repeating the call -- but maybe not waiting a min, but just a few seconds?

**Comment by avivajpeyi** [comment]
OOO interesting find! 

On running the notebook manually, I get a different error! 
`No data found for target "TIC 79395355".` 


```python
2022-03-08 23:43:10,106 - TESS-ATLAS - INFO - Downloading LightCurveData from MAST
No data found for target "TIC 79395355".

~/Documents/projects/tess/tess-atlas/src/tess_atlas/data/lightcurve_data.py in from_database(cls, tic, outdir)
     46 
     47         # Restrict to short cadence no "fast" cadence
---> 48         search = search[np.where(search.table["t_exptime"] == 120)]
     49 
     50         logger.info(

~/Documents/projects/tess/tess_venv/lib/python3.9/site-packages/astropy/table/table.py in __getitem__(self, item)
   1865     def __getitem__(self, item):
   1866         if isinstance(item, str):
-> 1867             return self.columns[item]
   1868         elif isinstance(item, (int, np.integer)):
   1869             return self.Row(self, item)

~/Documents/projects/tess/tess_venv/lib/python3.9/site-packages/astropy/table/table.py in __getitem__(self, item)
    244         """
    245         if isinstance(item, str):
--> 246             return OrderedDict.__getitem__(self, item)
    247         elif isinstance(item, (int, np.integer)):
    248             return list(self.values())[item]

KeyError: 't_exptime'
```

**Comment by avivajpeyi** [comment]
Search will have a len of zero -- there is no data, raise an error here

**Comment by avivajpeyi** [comment]
Manually run for a couple of the erroneous TOIs and document

**Comment by avivajpeyi** [comment]
restrict catalog based on if there is lk data

**Comment by avivajpeyi** [comment]
- [ ] Raise error on failed lk search
- [ ] Restrict TOI catalog based on avail lk data

**Comment by avivajpeyi** [comment]
From `lk` devs: 
> a TIC is not guaranteed to have lightcurve products available for a variety of reason

Maybe we can proceed by skipping the analysis for the ~2K TIC which do not have lightcurve data.

**Comment by avivajpeyi** [comment]
"Solved" in #200
